plugins/modules/parallels_desktop.py

```diff
@@ -0,0 +1,685 @@
+                format(duration=cycle_delay * wait_cycle),
+            )
+
+            try:
+                parallels_desktop_pid = self.get_parallels_pid()
+            except LookupError:
+                break
+            if parallels_desktop_pid != original_pid:
+                raise ParallelsDesktopModuleError(
+                    msg='Another `{app_name}` instance got re-spawned '
+                    'unexpectedly by an external process. Make sure not '
+                    'to start {app_name} while this module is running...'.
+                    format(app_name=PARALLELS_DESKTOP_APP_NAME),
+                    error_args={
+                        'Original {app_name} PID'.
+                        format(app_name=PARALLELS_DESKTOP_APP_NAME):
+                        original_pid,
+                        'New {app_name} PID'.
+                        format(app_name=PARALLELS_DESKTOP_APP_NAME):
+                        parallels_desktop_pid,
+                    },
+                )
+        else:
+            # Still running
+            raise TimeoutError(
+                'Timed out waiting for process PID {pid:d}.'.
+                format(pid=original_pid),
+            )
+
+    def kindly_ask_parallels_desktop_app_to_quit(self):  # type: () -> None
+        """Tell Parallels to quit via ``osascript``.
+
+        This function suppresses the following known situations:
+        * Parallels Desktop is showing the Update Available window during start
+        * The command is invoked under a user that is unable to look into the
+          apps being run under different users.
+        """
+        osascript_cmd_exec_args = (
+            self.osascript,
+            '-e',
+            'tell application "{app_name}" to quit'.
+            format(app_name=PARALLELS_DESKTOP_APP_NAME),
+        )
+        try:
+            self.run_with_raise(osascript_cmd_exec_args)
+        except CmdFailedError as cmd_err:
+            process_error_code = 1
+
+            user_cancelled_error_msg = (  # Stuck on Update Available on start
+                '40:44: execution error: Parallels Desktop got an error: '
+                'User canceled. (-128)'
+            )
+            app_isnt_running_error_msg = (  # Running under different user
+                '40:44: execution error: Parallels Desktop got an error: '
+                'Application isn’t running. (-600)'
+            )
+
+            expected_error_messages = {
+                app_isnt_running_error_msg,
+                user_cancelled_error_msg,
+            }
+
+            is_generic_error = cmd_err.error_args['rc'] == process_error_code
+            is_error_message_expected = (
+                cmd_err.error_args['stderr'] in expected_error_messages
+            )
+            if is_generic_error and is_error_message_expected:
+                pass
+
+            raise
+
+    def ensure_app_started(  # noqa: WPS231
```

Comment:
Self note: looks like I forgot to implement the check mode support in this method.